main.go

```diff
@@ -95,467 +56,15 @@ func printVersion(ctx context.Context, cmd *cli.Command, versionFlag bool) error
-			return testPlan, nil
-		}
-
-		debug.Printf("Test plan found. Identifier: %q", cfg.Identifier)
-		return *cachedPlan, nil
-	}
-
-	debug.Println("No test plan found, creating a new plan")
-	// If the cache is empty, create a new plan.
-	params, err := createRequestParam(ctx, cfg, files, *apiClient, testRunner)
-	if err != nil {
-		return handleError(err)
-	}
-
-	debug.Println("Creating test plan")
-	testPlan, err := apiClient.CreateTestPlan(ctx, cfg.SuiteSlug, params)
-
-	if err != nil {
-		return handleError(err)
-	}
-
-	// The server can return an "error" plan indicated by an empty task list (i.e. `{"tasks": {}}`).
-	// In this case, we should create a fallback plan.
-	if len(testPlan.Tasks) == 0 {
-		fmt.Println("⚠️ Error plan received, falling back to non-intelligent splitting. Your build may take longer than usual.")
-		testPlan = plan.CreateFallbackPlan(files, cfg.Parallelism)
-		return testPlan, nil
-	}
-
-	debug.Printf("Test plan created. Identifier: %q", cfg.Identifier)
-	return testPlan, nil
-}
-
-// createRequestParam generates the parameters needed for a test plan request.
-// For runners other than "rspec", it constructs the test plan parameters with all test files.
-// For the "rspec" runner, it filters the test files through the Test Engine API and splits the filtered files into examples.
-func createRequestParam(ctx context.Context, cfg config.Config, files []string, client api.Client, runner TestRunner) (api.TestPlanParams, error) {
-	testFiles := []plan.TestCase{}
-	for _, file := range files {
-		testFiles = append(testFiles, plan.TestCase{
-			Path: file,
-		})
-	}
-
-	// Splitting files by example is only supported for rspec runner & cucumber
-	if runner.Name() != "RSpec" && runner.Name() != "Cucumber" {
-		params := api.TestPlanParams{
-			Identifier:  cfg.Identifier,
-			Parallelism: cfg.Parallelism,
-			Branch:      cfg.Branch,
-			Runner:      cfg.TestRunner,
-			Tests: api.TestPlanParamsTest{
-				Files: testFiles,
-			},
-		}
-
-		// This is a workaround for the fact that the pytest-pants runner is not
-		// supported by the Test Engine API. For now, we use the pytest runner. At
-		// some point, there may be a difference between the two runners, but for
-		// now the response from the Test Engine API is the same for both runners.
-		if cfg.TestRunner == "pytest-pants" {
-			params.Runner = "pytest"
-		}
-
-		return params, nil
-	}
-
-	if cfg.SplitByExample {
-		debug.Println("Splitting by example")
-	}
-
-	// The SplitByExample flag indicates whether to filter slow files for splitting by example.
-	// Regardless of the flag's state, the API will still filter other files that need to be split by example, such as those containing skipped tests.
-	// Therefore, we must filter and split files even when SplitByExample is disabled.
-	testParams, err := filterAndSplitFiles(ctx, cfg, client, testFiles, runner)
-	if err != nil {
-		return api.TestPlanParams{}, err
+		// otherwise exit code 16
+		os.Exit(16)
```

Comment:
Almost all of the places where `logErrorAndExit` were being called previously were using exit code `16` except for the code that handled the exit status of the actual command that ran the tests https://github.com/buildkite/test-engine-client/blob/45db707f44cb53ef92da660edcc41bf3fb155f0c/main.go#L159-L162

This change lifts that logic here so this method only needs an `err` argument.